conan_provider.cmake

```diff
@@ -258,7 +258,7 @@ function(conan_version_check)
 endfunction()
 
 
-macro(conan_provide_dependency package_name)
+macro(conan_provide_dependency method package_name)
```

Comment:
Is this ``method`` arg unused?

Reply:
Yeah this is unused here as we only accept was method - this was a bit wrong before, as the first argument to the macro is the "method" which in this case should be `FIND_PACKAGE` according to the docs: https://cmake.org/cmake/help/latest/command/cmake_language.html#id2, https://cmake.org/cmake/help/latest/command/cmake_language.html#provider-commands. Since our provider only resolves calls to `find_package`, this argument is unused.

Before this PR this variable didn't contain the package name, but rather all the arguments to find_package (including the package name) - with this change, now `package_name` is now the very first argument to the `find_package` call, and `${ARGN}` contains all other arguments.